### PR TITLE
ref(dashboards): adds source prop dashboard widget

### PR DESCRIPTION
**Pull request by scefali**
This PR removes the `fromDiscover` and `fromLibrary` props from `DashboardWidgetModalOptions` and replaces it with the more generic `source` prop to denote where we are launching the dashboard widget. The reason for this change is so we can add a way of generating a widget from the issues page that will be similar to `fromDiscover`.

### CODE REVIEW

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 1]
```diff
@@ -67,8 +67,7 @@ export type DashboardWidgetModalOptions = {
   defaultTableColumns?: readonly string[];
   defaultTitle?: string;
   displayType?: DisplayType;
-  fromDiscover?: boolean;
-  fromLibrary?: boolean;
+  source: 'discoverv2' | 'dashboards' | 'library' | 'issueDetail';
```
Comment (edwardgou-sentry) [inline static/app/components/modals/addDashboardWidgetModal.tsx:70]: Just a nit, if we could store these as constants I think that would be an improvement!

**Review by edwardgou-sentry — APPROVED** [review]
Draft looks good to me so far, just a single comment about source type

File: static/app/components/modals/addDashboardWidgetModal.tsx (hunk 1 above)
Comment (scefali) [reply static/app/components/modals/addDashboardWidgetModal.tsx:70]: @edwardgou-sentry sure, I'll convert it

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 2]
```diff
@@ -141,13 +141,24 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
     }
     this.handleDefaultFields();
   }
 
+  get omitDashboardProp() {
+    // when opening from discover or issues page, the user selects the dashboard in the widget UI
+    return [
+      DashboardWidgetSource.DISCOVERV2,
+      DashboardWidgetSource.ISSUE_DETAILS,
+    ].includes(this.props.source || '');
```
Comment (evanpurkhiser) [inline static/app/components/modals/addDashboardWidgetModal.tsx:155]:
```suggestion
    ].includes(this.props.source ?? '');
```

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**